### PR TITLE
Changes for handling zero dollar and staff payment

### PR DESCRIPTION
**Pull request by pksumesh**
*Issue #:*
https://github.com/bcgov/entity/issues/1368
https://github.com/bcgov/entity/issues/1434

*Description of changes:*
- Added a new Internal Payment Service to handle all internal payments
- Changes for handling zero dollar fees.
- Adding new filing type OTFDR with charge as zero dollars 
- Changes for handling staff payments

*Checklist:*
I confirm that below checklist items are addressed in this pull request; (check the boxes applicable)
- [x] No lint errors
- [x] No test case failures and proper coverage for all modules/classes
- [ ] Updated the deployment configs for new environment variable(s)
- [ ] Updated the postman collection in entity repository (https://github.com/bcgov/entity/tree/master/api-e2e/postman) for e2e tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

**Comment by pksumesh** [comment]
```
(venv) Sumeshs-MacBook-Pro:pay-api aot$ make pylint
pylint --rcfile=setup.cfg \
	 --load-plugins=pylint_flask \
	 --disable=C0301,W0511 \
	 src/pay_api

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)


```

**Comment by pksumesh** [comment]
```
---------- coverage: platform darwin, python 3.7.1-final-0 -----------
Name                                            Stmts   Miss Branch BrPart  Cover
---------------------------------------------------------------------------------
src/pay_api/__init__.py                            37      0      0      0   100%
src/pay_api/exceptions/__init__.py                 13      0      0      0   100%
src/pay_api/factory/__init__.py                     1      0      0      0   100%
src/pay_api/factory/payment_system_factory.py      37      0     14      0   100%
src/pay_api/jwt_wrapper.py                         12      1      4      2    81%
src/pay_api/models/__init__.py                     18      0      0      0   100%
src/pay_api/models/audit.py                        12      0      0      0   100%
src/pay_api/models/base_model.py                   15      0      0      0   100%
src/pay_api/models/base_schema.py                  13      0     10      0   100%
src/pay_api/models/code_table.py                    5      0      0      0   100%
src/pay_api/models/corp_type.py                    13      0      0      0   100%
src/pay_api/models/db.py                            5      0      0      0   100%
src/pay_api/models/fee_code.py                     13      0      0      0   100%
src/pay_api/models/fee_schedule.py                 35      0      4      0   100%
src/pay_api/models/filing_type.py                  13      0      0      0   100%
src/pay_api/models/invoice.py                      54      0      8      0   100%
src/pay_api/models/payment.py                      41      0      6      0   100%
src/pay_api/models/payment_account.py              26      0      2      0   100%
src/pay_api/models/payment_line_item.py            30      0      0      0   100%
src/pay_api/models/payment_method.py               13      0      0      0   100%
src/pay_api/models/payment_system.py               13      0      0      0   100%
src/pay_api/models/payment_transaction.py          38      0      0      0   100%
src/pay_api/models/receipt.py                      23      0      2      0   100%
src/pay_api/models/status_code.py                  13      0      0      0   100%
src/pay_api/resources/__init__.py                  29      0      0      0   100%
src/pay_api/resources/apihelper.py                  7      2      0      0    71%
src/pay_api/resources/bcol_profile.py              21      0      2      0   100%
src/pay_api/resources/fee.py                       28      0      2      0   100%
src/pay_api/resources/invoice.py                   31      0      4      0   100%
src/pay_api/resources/invoice_receipt.py           36      0      4      0   100%
src/pay_api/resources/meta.py                      10      0      2      0   100%
src/pay_api/resources/ops.py                       18      0      4      0   100%
src/pay_api/resources/payment.py                   64      0     12      0   100%
src/pay_api/resources/status.py                    20      0      2      0   100%
src/pay_api/resources/transaction.py               63      0      6      0   100%
src/pay_api/schemas/__init__.py                     1      0      0      0   100%
src/pay_api/schemas/utils.py                       55     14     22      7    70%
src/pay_api/services/__init__.py                    6      0      0      0   100%
src/pay_api/services/auth.py                       18      0      8      1    96%
src/pay_api/services/base_payment_system.py        22      0      0      0   100%
src/pay_api/services/bcol_service.py               35      0      0      0   100%
src/pay_api/services/fee_schedule.py              106      0     10      0   100%
src/pay_api/services/internal_pay_service.py       27      0      0      0   100%
src/pay_api/services/invoice.py                   210      0     16      0   100%
src/pay_api/services/oauth_service.py              64      2     16      2    95%
src/pay_api/services/paybc_service.py             131      0     20      0   100%
src/pay_api/services/payment.py                   125      0      6      0   100%
src/pay_api/services/payment_account.py           110      0      6      0   100%
src/pay_api/services/payment_line_item.py         139      0      2      0   100%
src/pay_api/services/payment_service.py           156      0     40      0   100%
src/pay_api/services/payment_transaction.py       232      0     42      0   100%
src/pay_api/services/queue_publisher.py            31      0      2      0   100%
src/pay_api/services/receipt.py                    99      0      8      0   100%
src/pay_api/services/status_service.py             63      0     26      0   100%
src/pay_api/utils/__init__.py                       1      0      0      0   100%
src/pay_api/utils/auth.py                           3      0      0      0   100%
src/pay_api/utils/constants.py                     11      0      0      0   100%
src/pay_api/utils/enums.py                         29      0      0      0   100%
src/pay_api/utils/errors.py                        24      0      0      0   100%
src/pay_api/utils/handlers.py                       7      0      0      0   100%
src/pay_api/utils/logging.py                        9      0      2      0   100%
src/pay_api/utils/run_version.py                   10      0      2      0   100%
src/pay_api/utils/trace.py                          5      0      0      0   100%
src/pay_api/utils/util.py                           8      0      0      0   100%
src/pay_api/version.py                              2      0      0      0   100%
---------------------------------------------------------------------------------
TOTAL                                            2559     19    316     12    99%
Coverage XML written to file coverage.xml
```

File: pay-api/migrations/versions/bae02665e807_adding_zero_dollar_data.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        fee_schedule_table,
+        [
+            {'filing_type_code': 'OTFDR', 'corp_type_code': 'CP', 'fee_code': 'EN107', 'fee_start_date': date.today(),
+             'fee_end_date': None}
+        ]
+    )
+
+
+def downgrade():
+    op.execute('DELETE FROM fee_schedule WHERE filing_type_code = \'OTFDR\';')
```
Comment (sutherlanda) [inline pay-api/migrations/versions/bae02665e807_adding_zero_dollar_data.py]: This downgrade will fail if the codes are used anywhere (had same issue with auth).  One way to handle this is to a truncate cascade on the table.  It's a destructive operation, but downgrading the database is anyway.

File: pay-api/src/pay_api/services/internal_pay_service.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    def get_payment_system_url(self, invoice: Invoice, return_url: str):
+        """Return the payment system url."""
+        return None
+
+    def get_payment_system_code(self):
+        """Return INTERNAL as the system code."""
+        return PaymentSystem.INTERNAL.value
+
+    def create_account(self, name: str, account_info: Dict[str, Any]):
+        """Create account internal."""
```
Comment (sutherlanda) [inline pay-api/src/pay_api/services/internal_pay_service.py]: Is this implementation complete or is this just a placeholder?

Comment (pksumesh) [reply pay-api/src/pay_api/services/internal_pay_service.py]: Currently we don't create internal accounts, but this implementation is needed to satisfy the abstract super class.

File: pay-api/migrations/versions/bae02665e807_adding_zero_dollar_data.py (hunk 1 above)
Comment (pksumesh) [reply pay-api/migrations/versions/bae02665e807_adding_zero_dollar_data.py]: +1 for this.

**Pull request merged**